### PR TITLE
Closes #756: When opening a modal windows then focus should be set on dialog header

### LINKED ISSUE #756: When opening a modal windows then focus should be set on dialog header

### PR DESCRIPTION
**Pull request by sudmar**
Closes #756

### CODE REVIEW

**Comment by dbojanek** [comment]
On closing modal window focus is set to the last element on site which i think is unacceptable. 
We should discuss this with @wfenrich . 
Ideally focus should be set back to the element that opened the modal (for example when we do "Cite Dataset -> Ris" we got file download window. When we close the window focus is back on 'RIS' element).

**Comment by dbojanek** [comment]
Also: probably not related to this task but - when using the app with keyboard only, I was not able to enter options under Dataset->Edit->Permissions->[Options]. Also there were some issues tab and arrow keys to just enter the 'Permissions' element.

**Comment by sudmar** [comment]
@dbojanek the issue with the "return focus" when closing the modal was already discussed at least once on daily :P 
We (ie. me, Wojtek and Krzysiek) have agreed that this issue is not a part of this task and won't be fixed right now, because it might be difficult to achieve without some major rework to how the modals are handled. It's not a problem to focus on any given element on the page when closing the modal (such mechanism can be implemented easily), but there is no easy way to tell which element has triggered the modal window, and thus which element we should set the focus on.

tl;dr: this will be a separate issue and it will be dealt with at a later time.

**Review by dbojanek — APPROVED** [review]
